src/app.js

```diff
@@ -0,0 +1,64 @@
+import TodoCount from './TodoCount.js';
+import TodoList from './TodoList.js';
+import TodoInput from './TodoInput.js';
+import TodoFilter from './TodoFilter.js';
+import {FilterOptions} from './constants.js';
+
+function App() {
+  const $todoInput = document.querySelector('#new-todo-title');
+  const $todoList = document.querySelector('#todo-list');
+  const $todoCount = document.querySelector('.todo-count');
+  const $todoFilter = document.querySelector('.filters');
+  this.data = [];
+  this.filteredData = [];
+  this.selected = 'all';
+
+  this.addItem = (text) => {
+    this.data.push({
+      text: text,
+      isCompleted: false,
+    });
+
+    this.filterItems(this.selected);
+  };
+
+  this.removeItem = (index) => {
+    this.data.splice(index, 1);
+
+    this.filterItems(this.selected);
+  };
+
+  this.editItem = (index, text) => {
+      this.data[index].text = text;
+
+      this.todoList.updateItem(this.data);
+  }
+
+  this.filterItems = (type) => {
+    this.selected = type;
+
+    if (type === FilterOptions.ALL.type) {
+      this.filteredData = this.data;
+    } else if (type ===  FilterOptions.ACTIVE.type) {
+      this.filteredData = this.data.filter((item) => !item.isCompleted);
+    } else if (type ===  FilterOptions.COMPLETED.type) {
+      this.filteredData = this.data.filter((item) => item.isCompleted);
+    }
+    this.todoList.updateItem(this.filteredData);
+    this.todoCount.render(this.filteredData.length);
+  };
```

Comment:
```suggestion
  this.filterItems = (type) => {
    this.selected = type;
    this.todoList.updateItem(this.getFilteredItem());
    this.todoCount.render(this.getFilteredItem().length);
  };

  this.getFilteredItem = () =>
      this.data.filter(({ isCompleted }) =>
        (this.type === FilterOptions.ALL.type) ||
        (this.type === FilterOptions.COMPLETED.type && isCompleted) ||
        (this.type === FilterOptions.ACTIVE.type && !isCompleted));
```
이런식으로 분리해주면 좋을 것 같습니다!

Reply:
와 `filter`로 이렇게 구현할 수 있다는게 신기하네요,,! 👍